clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -606,30 +606,21 @@ def ConditionOp : CIR_Op<"condition", [
 // YieldOp
 //===----------------------------------------------------------------------===//
 
-def YieldOpKind_FT : I32EnumAttrCase<"Fallthrough", 2, "fallthrough">;
-
-def YieldOpKind : I32EnumAttr<
-    "YieldOpKind",
-    "yield kind",
-    [YieldOpKind_FT]> {
-  let cppNamespace = "::mlir::cir";
-}
-
 def YieldOp : CIR_Op<"yield", [ReturnLike, Terminator,
     ParentOneOf<["IfOp", "ScopeOp", "SwitchOp", "LoopOp", "AwaitOp",
                  "TernaryOp", "GlobalOp"]>]> {
-  let summary = "Terminate CIR regions";
+  let summary = "Represents the default branching behaviour of a region";
   let description = [{
-    The `cir.yield` operation terminates regions on different CIR operations:
-    `cir.if`, `cir.scope`, `cir.switch`, `cir.loop`, `cir.await`, `cir.ternary`
-    and `cir.global`.
-
-    Might yield an SSA value and the semantics of how the values are yielded is
-    defined by the parent operation.
+    The `cir.yield` operation terminates regions on different CIR operations,
+    and it is used to represent the default branching behaviour of a region.
+    Said branching behaviour is determinted by the parent operation. For
+    example, a yield in a `switch-case` region implies a fallthrough, while
+    a yield in a `cir.if` region implies a branch to the exit block, and so
+    on.
 
-    Optionally, `cir.yield` can be annotated with extra kind specifiers:
-    - `fallthrough`: execution falls to the next region in `cir.switch` case list.
-    Only available inside `cir.switch` regions.
+    In some cases, it might yield an SSA value and the semantics of how the
+    values are yielded is defined by the parent operation. For example, a
+    `cir.ternary` operation yields a value from one of its regions.
```

Comment:
Maybe it would be better to have a separate `cir.fallthrough` operation and leave `cir.yield` only to be used when the region yields control back to the parent. This would simplify the BranchOpTerminatorInterface definition for these operations and make the dialect more intelligible.